### PR TITLE
Windows Namespace & CNI Support

### PR DESCRIPTION
**Pull request by madhanrm**

### CODE REVIEW

**Comment by madhanrm** [comment]
I just have to get a PR out for getting vendor/github.com/opencontainers/runtime-tools/generate/generate.go into the main repo before we can vendor that in here.

File: pkg/config/config_windows.go [hunk 1]
```diff
@@ -23,10 +23,9 @@ import "github.com/containerd/containerd"
 // DefaultConfig returns default configurations of cri plugin.
 func DefaultConfig() PluginConfig {
 	return PluginConfig{
-		// TODO: JTERRY75 - Enable when we have CNI support.
 		CniConfig: CniConfig{
-			NetworkPluginBinDir:       "", // "E:\\containerd\\data\\cni\\bin",
-			NetworkPluginConfDir:      "", // "E:\\containerd\\data\\cni\\config",
+			NetworkPluginBinDir:       "C:\\containerd\\data\\cni\\bin",
```
Comment (jterry75) [inline pkg/config/config_windows.go]: I think the defaults should be under "C:\Program Files\containerd\data". That is where the true defaults are and then we can override these in the config file if we choose.

File: pkg/store/sandbox/netns_windows.go [hunk 2]
```diff
@@ -38,23 +38,24 @@ type NetNS struct {
...
+	}
+
+	return &NetNS{path: string(hcnNamespace.Id)}, nil
 }
 
 // LoadNetNS loads existing network namespace. It returns ErrClosedNetNS
 // if the network namespace has already been closed.
 func LoadNetNS(path string) (*NetNS, error) {
-	// TODO: JTERRY75 - Plugin cni
 	return &NetNS{restored: true, path: path}, nil
```
Comment (jterry75) [inline pkg/store/sandbox/netns_windows.go]: Do you want to query hcn in this case to make sure that network ID is still valid?

File: pkg/store/sandbox/netns_windows.go [hunk 3]
```diff
@@ -63,6 +64,11 @@ func (n *NetNS) Remove() error {
 	if n.restored {
 		n.restored = false
 	}
+	hcnNamespace, err := hcn.GetNamespaceByID(n.path)
+	if err != nil {
```
Comment (jterry75) [inline pkg/store/sandbox/netns_windows.go]: You might want to handle the `GetNamespaceByID` returning `NotFound`. This is also a success for `NetNS` based on the comments.

**Comment by jterry75** [comment]
@madhanrm - BTW made you a collaborator on this repo so you can easily merge things yourself too

**Comment by jterry75** [comment]
This looks go to me for a 1'st merge at least. Are you ready for me to merge this and give it a shot?

**Pull request merged**